src/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -89,6 +100,29 @@
 
     protected LibraryIdGenerator libraryIdGenerator = null; // this is initialized in buildSortedReadEndLists
 
+    private int getBarcodeValue(final SAMRecord record) {
```

Comment:
can this use the function getReadBarcodeValue(final SAMRecord record, final String tag) and avoid the code duplication?